src/main/java/ch/njol/skript/entity/ZombieVillagerData.java

```diff
@@ -82,7 +85,7 @@ protected boolean deserialize(final String s) {
 	@SuppressWarnings("null")
 	@Override
 	public void set(final ZombieVillager e) {
-		profession = e.getVillagerProfession();
+		e.setVillagerProfession(profession);
```

Comment:
this seems like different functionality, I think before it updated the `ZombieVillagerData` object based on `e` but this is updating `e` based on `ZombieVillagerData#profession`. is that on purpose?

Reply:
This method is used to set the zombie villagers profession.
Im not sure if it was written wrong before or what, but it didn't work. This fixes it.